fdbrpc/FlowTests.actor.cpp

```diff
@@ -1435,19 +1437,28 @@ TEST_CASE("/flow/flow/PromiseStream/move") {
 		stream.send(namedTracker1);
 		stream.send(namedTracker2);
 		{
-			Tracker t = waitNext(stream.getFuture());
-			ASSERT(!t.moved);
+			state Tracker copiedTracker = waitNext(stream.getFuture());
+			ASSERT(!copiedTracker.moved);
 			// must copy onto queue
-			ASSERT(t.copied == 1);
+			ASSERT(copiedTracker.copied == 1);
 		}
-		choose {
-			when(Tracker t = waitNext(stream.getFuture())) {
-				ASSERT(!t.moved);
-				// must copy onto queue
-				ASSERT(t.copied == 1);
-			}
+		{
+			Tracker copiedTracker = waitNext(stream.getFuture());
+			ASSERT(!copiedTracker.moved);
+			// must copy onto queue
+			ASSERT(copiedTracker.copied == 1);
 		}
 	}
 
 	return Void();
 }
+
+TEST_CASE("/flow/flow/PromiseStream/move2") {
+	PromiseStream<Tracker> stream;
+	stream.send(Tracker{});
+	Tracker tracker = waitNext(stream.getFuture());
+	Tracker movedTracker = std::move(tracker);
```

Comment:
Can you also `ASSERT(tracker.moved);` here, even if only for clarity, because (if I understand correctly) if the new `T &&` callback was used then `moved` would be true after this line, but if not then it would be false? 

Looking at `Tracker`, it seems that for either callback `movedTracker.move` would be false so that may not be telling, but I do think `moveTracker.copied == 0` is only true if the expected thing has happened.  But in case that's somehow wrong, checking `tracker.moved` seems like a good idea anyway.